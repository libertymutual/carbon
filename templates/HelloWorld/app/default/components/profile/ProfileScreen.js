import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import CounterActions from '../../actions/CounterActions';

import * as PresentationConstants from '../../constants/PresentationConstants';

/**
 * Profile Screen
 * @extends {React.Component}
 */
class ProfileScreen extends React.Component {
  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    return (<ScrollView>
      <Text style={styles.profileScreenHeader}>{PresentationConstants.PROFILE_WELCOME}</Text>
      <Text style={styles.profileScreenText}>{PresentationConstants.PROFILE_INTRO}</Text>
      <Text
        style={styles.profileScreenText}
      >{PresentationConstants.PROFILE_COUNTER_VALUE}{this.props.counter}</Text>
      <Button
        onPress={this.props.counterActions.increment}
        title={PresentationConstants.PROFILE_INCREMENT_COUNTER}
      />
      <Button
        onPress={this.props.counterActions.decrement}
        title={PresentationConstants.PROFILE_DECREMENT_COUNTER}
      />
      <Button
        onPress={this._navigateBack}
        title={PresentationConstants.GO_BACK}
      />
    </ScrollView>);
  }

  /**
   * Navigates back to the previous screen
   * @private
   */
  _navigateBack = () => {
    this.props.navigation.goBack();
  }

}

/**
 * Navigation Options
 */
ProfileScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_PROFILE_SCREEN,
};

/**
 * Prop Types
 */
ProfileScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  counter: PropTypes.number.isRequired,
  counterActions: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @param {Object} state Redux State Object
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps(state) {
  return {
    counter: state.counter,
  };
}

/**
 * Maps Dispatch Actions To Props
 * @param {function} Dispatch function
 * @returns {Object) Object representing the actions
 * that should be exposed to the component via props
 */
function mapDispatchToProps(dispatch) {
  return {
    counterActions: bindActionCreators(CounterActions, dispatch),
  };
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(ProfileScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  profileScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  profileScreenText: {
    margin: 14,
  },
});
