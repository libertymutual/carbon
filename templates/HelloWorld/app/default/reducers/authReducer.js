import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
  isLoggedIn: false,
  username: "",
  password: "",
  authUrl: "",
  clientId: "",
  clientSecret: "",
  configName: "",
  validatorId: "",
}

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function authReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.LOGIN:
      return { ...state, isLoggedIn: true };
    case ActionTypes.LOGOUT:
      return { ...state, isLoggedIn: false };
    case ActionTypes.AUTH_UPDATE_AUTH_URL:
      return { ...state, authUrl: action.authUrl };
    case ActionTypes.AUTH_UPDATE_CLIENT_ID:
      return { ...state, clientId: action.clientId };
    case ActionTypes.AUTH_UPDATE_CLIENT_SECRET:
      return { ...state, clientSecret: action.clientSecret };
    case ActionTypes.AUTH_UPDATE_CONFIG_NAME:
      return { ...state, configName: action.configName };
    case ActionTypes.AUTH_UPDATE_PASSWORD:
      return { ...state, password: action.password };
    case ActionTypes.AUTH_UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.AUTH_UPDATE_VALIDATOR_ID:
      return { ...state, validatorId: action.validatorId };
    default:
      return state;
  }
}
