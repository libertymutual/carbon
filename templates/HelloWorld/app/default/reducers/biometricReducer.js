import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
    username: "TestUsername",
    password: "TestPassword",
    credentialsSaved: false
};

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function biometricReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.UPDATE_PASSWORD:
      return { ...state, password: action.password };  
    case ActionTypes.SAVE_CREDENTIALS:
      return { ...state, credentialsSaved: true };
    case ActionTypes.DELETE_CREDENTIALS:
      return { ...state, credentialsSaved: false };
    default:
      return state;
  }
}
