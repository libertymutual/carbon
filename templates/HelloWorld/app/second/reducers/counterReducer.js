import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = 0;

/**
 * Reducer handling counter related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function countReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.INCREMENT:
      return state + 1;
    case ActionTypes.DECREMENT:
      return state - 1;
    default:
      return state;

  }
}
