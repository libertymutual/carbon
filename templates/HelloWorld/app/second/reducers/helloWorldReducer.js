import * as ActionTypes from '../actions/ActionTypes';
import * as Constants from '../constants/Constants';

/**
 * Initial Reducer State
 */
export const initialState = {
  state: Constants.HELLO,
};

/**
 * Reducer handling hello world related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function helloWorldReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.HELLO:
      return { ...state, state: Constants.HELLO };
    case ActionTypes.GOODBYE:
      return { ...state, state: Constants.GOODBYE };
    default:
      return state;
  }
}
