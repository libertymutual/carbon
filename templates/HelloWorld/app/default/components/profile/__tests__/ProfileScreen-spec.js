
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ProfileScreen from '../ProfileScreen';
import { initialState } from '../../../reducers/counterReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let store;

describe('ProfileScreen tests', () => {
  beforeEach(() => {
    // Sets up a new store
    store = mockStore({
      counter: initialState
    });
  });

  test('Profile Screen renders correctly', () => {
    const tree = renderer.create(
      <ProfileScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

});
