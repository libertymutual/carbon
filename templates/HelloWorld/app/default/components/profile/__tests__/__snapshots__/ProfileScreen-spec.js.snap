// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen tests Profile Screen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Welcome To The Default Feature Profile Screen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      This sample page allows you to test updating a persistent store, the value of which will be retained between app restarts
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      Counter Value:
      0
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Increment Counter
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Decrement Counter
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
